rust_src/src/symbols.rs

```diff
@@ -58,3 +59,21 @@ fn symbol_function(object: LispObject) -> LispObject {
 fn symbol_plist(object: LispObject) -> LispObject {
     object.as_symbol_or_error().get_plist()
 }
+
+/// Intern a string (e.g. create a symbol from it, if it does not exist
+/// already).
+///
+/// This is an implementation equivalent of the C version.
+pub unsafe fn intern(s: *const libc::c_char) -> LispObject {
+    LispObject::from_raw(intern_1(s, libc::strlen(s) as libc::ptrdiff_t))
+}
+
+/// Intern (e.g. create a symbol for) a `&'static str`, avoiding
+/// allocation. This macro is inherently unsafe: if the string contains internal
+/// NULL bytes (`\0`), we will violate memory safety.
```

Comment:
Why would it violate memory safety? It would just use the shorter string since you're using `strlen` to determine the length.